### DISCUSSION TITLE
I have been trying swipe left or swipe right action for long time without any positive response in my emulator. Is there any solution for this without using touchactions?

### DISCUSSION THREAD

**Discussion by Balakumar-14** [appium/appium#17010] [General]
Swipe up, swipe down and swipe left actions are well executable. But, swipe right is getting failed.
While running with this action, no error produces. And, action too produced.
I a trying without using touchactions module. Because, they will be depricated soon.
So, help me with my code, as  I am a beginner. 
Thanks in advance.

Code I have tried:
        `public void swipeRight() {
        JavascriptExecutor js = driver;
        HashMap<String,Object> map = new HashMap<>();
        map.put("direction","left");
        map.put("percent", 1.0);
        map.put("left", 0);
        map.put("top", 1500);
        map.put("width", 1000);
        map.put("height", 0);
        map.put("speed", 500);
        js.executeScript("mobile: swipeGesture",map);
    }

    public void swipe_Left() {
        JavascriptExecutor js = driver;
        HashMap<String,Object> map = new HashMap<>();
        map.put("direction","right");
        map.put("percent", 1.0);
        map.put("left", 0);
        map.put("top", 1500);
        map.put("width", 1000);
        map.put("height", 0);
        map.put("speed", 500);
        js.executeScript("mobile: swipeGesture",map);
    }

    public void swipeDown()
    {
        JavascriptExecutor js = driver;
        HashMap<String,Object> map = new HashMap<>();
        map.put("direction","up");
        map.put("percent",1.0);
        map.put("left",200);
        map.put("top",500);
        map.put("width",0);
        map.put("height",1100);
        map.put("speed", 500);
        js.executeScript("mobile: swipeGesture",map);
    }

    public void swipeUp() {
        JavascriptExecutor js = driver;
        HashMap<String,Object> map = new HashMap<>();
        map.put("direction","down");
        map.put("percent", 1.0);
        map.put("left", 300);
        map.put("top", 700);
        map.put("width", 0);
        map.put("height", 1100);
        map.put("speed", 500);
        js.executeScript("mobile: swipeGesture",map);
    }`

**Comment by KazuCocoa** [comment]
Asking in https://discuss.appium.io/ may help to ask a general question as the first place.
